crates/ty_python_semantic/resources/mdtest/annotations/self.md

```diff
@@ -208,6 +301,53 @@ class MyMetaclass(type):
         return super().__new__(cls)
 ```
 
+## Explicit Annotation Overrides Implicit `Self`
+
+If the first parameter is explicitly annotated, that annotation takes precedence over the implicit
+`Self` treatment.
+
+```toml
+[environment]
+python-version = "3.11"
+```
+
+```py
+class Explicit:
+    # TODO: Should warn the user if self is overriden with a type that is not subtype of the class
```

Comment:
This should be supertype instead of subtype?

https://pyright-play.net/?pyrightVersion=1.1.405&code=MYGwhgzhAECCBcBYAUNN0AmBTAZtARmBgBQRYg7zQBEAwtQJTQC0AfNAHID2AdlkqnRCAdKJQpQkGLWKwGVUcKA